modules/packages/LinearAlgebra.chpl

```diff
@@ -773,6 +773,97 @@ proc _matmatMult(A: [?Adom] ?eltType, B: [?Bdom] eltType)
   return C;
 }
 
+
+pragma "no doc"
+/* Iterator for blocking up dimension wrt window */
+private iter block(indexes, window : int) {
+  var lowVal = indexes.low;
+  const stride = indexes.stride;
+  if indexes.stridable{
+    while (true) {
+      var highVal = lowVal + window * stride - 1;
+      if (highVal >= indexes.high) {
+        yield lowVal..indexes.high by stride;
+        break;
+      } else {
+        yield lowVal..highVal by stride;
+      }
+      lowVal = highVal + 1;
+    }
+  } else {
+    while (true) {
+      var highVal = lowVal + window - 1;
+      if (highVal >= indexes.high) {
+        yield lowVal..indexes.high;
+        break;
+      } else {
+        yield lowVal..highVal;
+      }
+      lowVal = highVal + 1;
+    }
+  }
+}
+
+
+pragma "no doc"
+/* Distributed matrix-matrix multiplication */
+proc _matmatMult(A : [?Adom] ?eltType, B : [?Bdom] eltType, in window : int = -1)
+  where isDistributed(A) || isDistributed(B)
+{
+  ref targetLocales = A.targetLocales();
+
+  if A.shape(1) != B.shape(0) {
+    halt("Array dimensions don't match.\n Trying to multiply arrays of dimensions ", 
+         A.shape, " ", B.shape);
+  }
+
+  const commonDim = Adom.dim(1);
+  if window < 1 {
+    window = commonDim.size;
+  }
+
+  ref Bref = B.reindex(commonDim, Bdom.dim(1));
```

Comment:
I don't think this is used anywhere.

Reply:
Yep, it rolled over from the older version of the implementation